Add button component and update alert

helrond opened the pull request:
Adds a Button component with an associated test. This is largely based off of the same component in DIMES, with the icons removed. 

A few things to review/pay attention to:
- Some of these properties may be unnecessary for this application but maybe that's something we can revisit later?
- This component just returns directly (as opposed to the Alert component does). In general I think direct returns are preferred because of concision but see if you agree!
- I added a PropTypes array to this component, which I think is helpful in terms of understanding how to use the component and may be good to implement in other components. LMK what you think!

Fixes #6

helrond committed:
ignoring _site dir

helrond committed:
add button component

helrond committed:
add buttom tests

helrond committed:
update proptypes

HaSistrunk commented on app/__tests__/Button.test.js:
```diff
@@ -0,0 +1,36 @@
+
+describe('Button Component', () => {
+
+  // Test if the table renders columns and data
```
```suggestion
```

HaSistrunk commented on .gitignore:
```diff
@@ -17,6 +17,9 @@
 /out/
 
+# development
+/_site
```
I don't see this directory. Where is that coming from?

helrond replied:
Ah you know what that's a holdover from the previous architecture. Removing.

HaSistrunk commented on app/components/Button.js:
```diff
@@ -0,0 +1,37 @@
+    'black']).isRequired,
+  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg']).isRequired,
+  handleClick: PropTypes.func.isRequired,
+  ariaLabel: PropTypes.string.isRequired,
```
```suggestion
  ariaLabel: PropTypes.string,
```

HaSistrunk reviewed (commented):
Cool! Re: propTypes and direct return - that approach makes sense to me. Do we want to align the alert component with that approach in this same PR?

helrond committed:
Update app/components/Button.js

Co-authored-by: Hannah Sistrunk <hsistrunk@rockarch.org>

helrond committed:
Update app/__tests__/Button.test.js

Co-authored-by: Hannah Sistrunk <hsistrunk@rockarch.org>

helrond committed:
undoing unnecessary addition to gitignore

helrond committed:
formatting

helrond committed:
remove PropTypes

helrond committed:
Direct return plus tests

helrond commented:
OK I've done a couple of things here:
- converted the Alert component to a direct return and added a test
- Removed PropTypes since that is [now deprecated](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#removed-proptypes-and-defaultprops)

HaSistrunk commented on app/components/Button.js:
```diff
@@ -0,0 +1,14 @@
+const Button = props => (
+  <button
+    type={props.type}
+    className={`btn btn--${props.color} btn--${props.size}`}
+    onClick={props.handleClick}
+    aria-label={props.ariaLabel}
+    aria-haspopup={props.ariaHasPopup}
+    aria-expanded={props.ariaExpanded}
+    aria-pressed={props.ariaPressed}
+    disabled={props.disabled}>
+      {props.label}
+  </button>)
+
+export default Button
```
Following the pattern of the badge and alert components, wouldn't the structure be:
```
const Button = ({
      type,
      color,
      size,
      handleClick,
      label,
      ariaLabel
      ariaHasPopup,
      ariaExpanded,
      ariaPressed,
      disabled
    }) => (
      <button
        type={type}
        className={`btn btn--${color} btn--${size}`}
        onClick={handleClick}
        aria-label={ariaLabel}
        aria-haspopup={ariaHasPopup}
        aria-expanded={ariaExpanded}
        aria-pressed={ariaPressed}
        disabled={disabled}
      >
        {label}
      </button>
    );

helrond replied:
It's _kind_ of a preference thing - there's no hard rule about when you expand props and when you don't, but generally speaking when there are this many props I don't expand them, just for concision's sake.

HaSistrunk committed:
Merge branch 'react-next' into issue-6

HaSistrunk committed:
lint alert, button, and their tests
